### PR TITLE
manage subscription changes

### PR DESCRIPTION
**Pull request by tejassonar**
- Manage subscription changes
- Edit Paypal Subscription implemented
- Pause donation even for trailing donations

Refer to the following spreadsheet to check the conditions:
https://docs.google.com/spreadsheets/d/1WkSoUfT2iv059Lee0YGnTCR8Q8tAUsvd9IfsQTGvslU/edit#gid=0

How to test:
1. Create some Donation Subscriptions from donate with plant-for-the-Planet (try paypal, sepa, sofort etc.
    for Paypal - https://donate-with-planet-git-feature-paypalsubscription-planetapp.vercel.app/
    for Sepa - https://donate-with-planet-git-feature-sepasubscription-planetapp.vercel.app/
    for others - paydev.pp.eco
2. Try editing, cancelling, pausing donations from the Webapp Dashboard
3. Report any unusual behaviour 

Note: Paypal will take some time to update the donation. Refresh the page 3-4 times, the changes should be updated.

### CODE REVIEW

**Comment by norbertschuler** [comment]
I added a patch fixing a 400 error if sending an empty request after editing a subscription without doing any change. And fix the eslint error at the same place.

**Comment by norbertschuler** [comment]
- [ ] If editing and changing the amount of a Paypal subscription on localhost:3000 in my Chrome browser on MacOS and new tab for Paypal is opened to confirm the new amount for the subscription and after confirming I get the following message. Also the subscriptions in the tab where I edited the Paypal subscription does not get updated (e.g. by any callback):

Bildschirmfoto 2022-02-19 um 12 50 21

**Comment by norbertschuler** [comment]
> Note: Paypal will take some time to update the donation. Refresh the page 3-4 times, the changes should be updated.

We need to notify the user about this as otherwise we will think his change has failed!

**Comment by norbertschuler** [comment]
- [ ] API calls on staging are very slow. If they are also slow on production, we should add a loading indicator.

**Comment by norbertschuler** [comment]
- [ ] If you have two subscriptions (one with Stripe and one with Paypal) there is a bug in the way that if you want to cancel the Paypal subscription immediately (no other option is possible for Paypal in the cancel dialog) the API call is still done with `cancellationType: "custom-date"` and `cancellationDate: null` which of course produced a 400 error at the backend. It probably kept the wrong values from a former cancellation I did for the Stripe subscription.
 
Bildschirmfoto 2022-02-19 um 13 03 40
 with the other option:

**Review by norbertschuler — DISMISSED** [review]
Still some bugs.

**Comment by delightedMatt** [comment]
I tested:
**SEPA**
- Pausing
  - For Date
  - For intermediary
- Editing
  - Amount
  - Date
- Cancelling
  - On a Date
  - Immediately
  
  Everything works!

**Comment by delightedMatt** [comment]
**PayPal**
- Pausing
  - For Date
  - For intermediary

- Editing
  - Amount
   - **==> Leads to PayPal for Confirmation. Confirming does not work at the moment.**
   - "Things don't appear to be working at the moment. Please try again later."
  - Date

- Cancelling
  - On a Date
  - Immediately

There are two label bugs:
canceldonation
notetowait

File: src/features/user/Account/CancelModal.tsx [hunk 1]
```diff
@@ -52,11 +61,13 @@ export const CancelModal = ({
       .then((res) => {
         console.log(res, 'Response');
         handleCancelModalClose();
+        fetchRecurrentDonations();
       })
       .catch((err) => {
         console.log(err, 'Error');
```
Comment (sarveshpro) [inline src/features/user/Account/CancelModal.tsx:67]: avoid consoling error object

**Review by delightedMatt — APPROVED** [review]
Reviewed entire Functionality for SEPA and PayPal and everything worked as intended.

**Review by sarveshpro — APPROVED** [review]
LGFM ✅

**Pull request merged**